services/app-web/src/components/FileUpload/FileUpload.tsx

```diff
@@ -33,6 +33,8 @@ export type FileUploadProps = {
     deleteFile: (key: string) => Promise<void>
     onFileItemsUpdate: ({ fileItems }: { fileItems: FileItemT[] }) => void
     isContractOnly?: boolean
+    shouldValidate?: boolean
```

Comment:
This is one of the places I was thinking of when I said we need to consider our error handling.  Maybe React Context for error state?  We should chat.

Reply:
I think that if we passed the file view component into the file uploads component that would go a long way to keeping the FileUpload component clean. 

I also think it’s a good point about “shouldValidate” not neccecarily belonging inside FileUpload. Let the Documents.tsx page manage when to display errors, if possible. the uploads componenet just reports the errors it sees upwards. Maybe with us having more detailed errors in the errors view, we can get rid of the error display that is inline in the documents component to simplify things?

I’m totally happy with us adding a refactor ticket, this is a tricky component and it could be fun for us all to collaberate on it.

Reply:
@haworku and I chatted about this (offline? online? in Zoom) and we're going to leave it as-is, because we both agree that a refactor of error handling across the various components that use fileUpload is necessary.